aws-lc-rs/src/tlsprf.rs

```diff
@@ -0,0 +1,295 @@
+        ];
+
+        let secret = super::Secret::new(super::P_SHA256, TLS_SECRET).expect("secret created");
+
+        let mut output = vec![0u8; TLS_OUTPUT_P_SHA256.len()];
+
+        // The AWS-LC KAT annoyingly includes the null terminator in its KAT, so using CString here to handle that for us.
+        let label = CString::new(TLS_LABEL).expect("failed to create CString");
+
+        secret
+            .derive_with_seed_concatination(
+                label.as_bytes_with_nul(),
+                TLS_SEED1,
+                TLS_SEED2,
+                &mut output,
+            )
+            .unwrap();
+
+        assert_eq!(TLS_OUTPUT_P_SHA256, output.as_slice());
+
+        let mut output = vec![0u8; TLS_OUTPUT_P_SHA256.len()];
+
+        let mut seed = Vec::<u8>::with_capacity(TLS_SEED1.len() + TLS_SEED2.len());
+        seed.extend(TLS_SEED1);
+        seed.extend(TLS_SEED2);
+
+        let secret = super::Secret::new(super::P_SHA256, TLS_SECRET).expect("secret created");
+        secret
+            .derive(label.as_bytes_with_nul(), &seed, &mut output)
+            .unwrap();
+
+        assert_eq!(TLS_OUTPUT_P_SHA256, output.as_slice());
+    }
+
+    #[test]
+    fn sha256() {
+        // KAT Sourced from https://csrc.nist.gov/Projects/cryptographic-algorithm-validation-program/Component-Testing
+
+        const SECRET: &[u8] = &[
+            0xf8, 0x93, 0x8e, 0xcc, 0x9e, 0xde, 0xbc, 0x50, 0x30, 0xc0, 0xc6, 0xa4, 0x41, 0xe2,
+            0x13, 0xcd, 0x24, 0xe6, 0xf7, 0x70, 0xa5, 0x0d, 0xda, 0x07, 0x87, 0x6f, 0x8d, 0x55,
+            0xda, 0x06, 0x2b, 0xca, 0xdb, 0x38, 0x6b, 0x41, 0x1f, 0xd4, 0xfe, 0x43, 0x13, 0xa6,
+            0x04, 0xfc, 0xe6, 0xc1, 0x7f, 0xbc,
+        ];
+        const LABEL: &[u8] = b"master secret";
+        const SEED1: &[u8] = &[
+            0x36, 0xc1, 0x29, 0xd0, 0x1a, 0x32, 0x00, 0x89, 0x4b, 0x91, 0x79, 0xfa, 0xac, 0x58,
+            0x9d, 0x98, 0x35, 0xd5, 0x87, 0x75, 0xf9, 0xb5, 0xea, 0x35, 0x87, 0xcb, 0x8f, 0xd0,
+            0x36, 0x4c, 0xae, 0x8c,
+        ];
+        const SEED2: &[u8] = &[
+            0xf6, 0xc9, 0x57, 0x5e, 0xd7, 0xdd, 0xd7, 0x3e, 0x1f, 0x7d, 0x16, 0xec, 0xa1, 0x15,
+            0x41, 0x58, 0x12, 0xa4, 0x3c, 0x2b, 0x74, 0x7d, 0xaa, 0xaa, 0xe0, 0x43, 0xab, 0xfb,
+            0x50, 0x05, 0x3f, 0xce,
+        ];
+        const EXPECT: &[u8] = &[
+            0x20, 0x2c, 0x88, 0xc0, 0x0f, 0x84, 0xa1, 0x7a, 0x20, 0x02, 0x70, 0x79, 0x60, 0x47,
+            0x87, 0x46, 0x11, 0x76, 0x45, 0x55, 0x39, 0xe7, 0x05, 0xbe, 0x73, 0x08, 0x90, 0x60,
+            0x2c, 0x28, 0x9a, 0x50, 0x01, 0xe3, 0x4e, 0xeb, 0x3a, 0x04, 0x3e, 0x5d, 0x52, 0xa6,
+            0x5e, 0x66, 0x12, 0x51, 0x88, 0xbf,
+        ];
+
+        let mut output = vec![0u8; EXPECT.len()];
+
+        let secret = Secret::new(P_SHA256, SECRET).expect("secret created");
+
+        secret
+            .derive_with_seed_concatination(LABEL, SEED1, SEED2, &mut output)
+            .expect("dervice successful");
```

Comment:
```suggestion
            .expect("derive successful");
```
Some others below as well.